toplevel/toploop.ml

```diff
@@ -60,23 +65,30 @@ let use_output ppf command =
          let ic = open_in_bin fn in
          Misc.try_finally ~always:(fun () -> close_in ic)
            (fun () ->
-              use_channel ppf ~wrap_in_module:false ic "" "(command-output)")
+            let lexbuf = (Lexing.from_channel ic) in
+            use_channel ppf ~wrap_in_module:false lexbuf "" "(command-output)")
        | n ->
          fprintf ppf "Command exited with code %d.@." n;
          false)
 
-let use_file ppf ~wrap_in_module name =
-  match name with
-  | "" ->
-    use_channel ppf ~wrap_in_module stdin name "(stdin)"
-  | _ ->
-    match Load_path.find name with
+let use_file ppf ~wrap_in_module usage =
+  match usage with
+  | {tag=Stdin} ->
+    let lexbuf = Lexing.from_channel stdin in
+    use_channel ppf ~wrap_in_module lexbuf "" "(stdin)"
+  | {tag=String; value } ->
+    let lexbuf = Lexing.from_string value in
+    use_channel ppf ~wrap_in_module lexbuf "" "(stdin)"
+  | { value; _} ->
```

Comment:
I would prefer a `tag = File` pattern here to explicitly handle all three cases.

Reply:
fixed